addons/point_of_sale/controllers/main.py

```diff
@@ -14,15 +14,17 @@ class PosController(http.Controller):
     @http.route('/pos/web', type='http', auth='user')
     def pos_web(self, **k):
         # if user not logged in, log him in
-        pos_sessions = request.env['pos.session'].search([
+        pos_sessions = request.env['pos.session'].sudo().search([
             ('state', '=', 'opened'),
             ('user_id', '=', request.session.uid),
             ('rescue', '=', False)])
         if not pos_sessions:
             return werkzeug.utils.redirect('/web#action=point_of_sale.action_client_pos_menu')
         pos_sessions.login()
+        session_info = request.env['ir.http'].session_info()
+        session_info['user_context']['allowed_company_ids'] = [pos_sessions.company_id.id]
```

Comment:
- you could use `.ids` to get a list directly
- might be worth a small comment above to explain that we want to restrict to a single company all the time for PoS, regardless of how many companies are currently visible in the web client